pkg/entpgx/pgxpool_dialect_driver.go

```diff
@@ -0,0 +1,258 @@
+package entpgx
+
+import (
+	"context"
+	stdsql "database/sql"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/dialect/sql"
+	"fmt"
+	"github.com/jackc/pgx/v5"
+	"github.com/kong/pg-aurora-client/pkg/pool"
+)
+
+func NewPgxPoolDriver(pool pool.PGXConnPool) dialect.Driver {
+	return &EntPgxpoolDriver{
+		pool: pool,
+	}
+}
+
+type EntPgxpoolDriver struct {
```

Comment:
Would have been nice to do a compilation check to ensure this implements the proper interface.
https://go.dev/doc/effective_go#blank_implements